exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/retry/RetryInterceptor.java

```diff
@@ -38,44 +47,72 @@ public RetryInterceptor(RetryPolicy retryPolicy, Function<Response, Boolean> isR
   RetryInterceptor(
       RetryPolicy retryPolicy,
       Function<Response, Boolean> isRetryable,
+      Function<IOException, Boolean> isRetryableException,
       Sleeper sleeper,
       BoundedLongGenerator randomLong) {
     this.retryPolicy = retryPolicy;
     this.isRetryable = isRetryable;
+    this.isRetryableException = isRetryableException;
     this.sleeper = sleeper;
     this.randomLong = randomLong;
   }
 
   @Override
   public Response intercept(Chain chain) throws IOException {
-    Response response = chain.proceed(chain.request());
-    if (!Boolean.TRUE.equals(isRetryable.apply(response))) {
-      return response;
-    }
+    Response response = null;
+    IOException exception = null;
+    int attempt = 0;
     long nextBackoffNanos = retryPolicy.getInitialBackoff().toNanos();
-    // Made an attempt already, starting with the 2nd.
-    for (int attempt = 2; attempt <= retryPolicy.getMaxAttempts(); attempt++) {
-      // https://github.com/grpc/proposal/blob/master/A6-client-retries.md#exponential-backoff
-      // Careful - the document says "retry attempt" but our maxAttempts includes the original
-      // request. The numbers are tricky but the unit test should be more clear.
-      long upperBoundNanos = Math.min(nextBackoffNanos, retryPolicy.getMaxBackoff().toNanos());
-      long backoffNanos = randomLong.get(upperBoundNanos);
+    do {
+      if (attempt > 0) {
+        // Compute and sleep for backoff
+        // https://github.com/grpc/proposal/blob/master/A6-client-retries.md#exponential-backoff
+        long upperBoundNanos = Math.min(nextBackoffNanos, retryPolicy.getMaxBackoff().toNanos());
+        long backoffNanos = randomLong.get(upperBoundNanos);
+        nextBackoffNanos = (long) (nextBackoffNanos * retryPolicy.getBackoffMultiplier());
+        try {
+          sleeper.sleep(backoffNanos);
+        } catch (InterruptedException e) {
+          Thread.currentThread().interrupt();
+          break; // Break out and return response or throw
+        }
+        // Close response from previous attempt
+        if (response != null) {
+          response.close();
+        }
+      }
 
+      attempt++;
       try {
-        sleeper.sleep(backoffNanos);
-      } catch (InterruptedException e) {
-        Thread.currentThread().interrupt();
-        return response;
+        response = chain.proceed(chain.request());
+      } catch (IOException e) {
+        throttlingLogger.log(
```

Comment:
I'm pretty sure callers will log this as well so we shouldn't log here

Reply:
Thought was that users may be interested in seeing log information describing why retrys occur. But if we wanted to log that information, we'd want to do it for more than just the case where `IOException` is thrown. Will leave it for possible future enhancement.